### PR TITLE
feat(shell-header-nav): nav links

### LINKED ISSUE #1095: Add menu support in Header
https://github.com/ibm/carbon-components/issues/1089 and

image

Latest sketch file: https://ibm.box.com/s/4e4zrz6mlo0jpr8lw4vimwlxn4vc59cy (internal)

### PR DESCRIPTION
**Pull request by aledavila**
Closes #1095 

Staging link: http://shell-nav-header-boring-otter.mybluemix.net/component/ui-shell--default

*Ignoring the Nav link font-weight difference. Bug has been submitted*

### CODE REVIEW

**Review by joshblack — CHANGES_REQUESTED** [review]
Could we use one arrow icon SVG and then have it rotate on hover?

Also noticed that when tabbing through, the menu will  disappear even though I should be on a nested link potentially?

Final note was that the font weights seem to be different for menus, is that intended?

File: src/components/ui-shell/ui-shell.config.js [hunk 1]
```diff
@@ -37,6 +37,26 @@ const header = {
   ],
 };
 
+const headernav = {
```
Comment (joshblack) [inline src/components/ui-shell/ui-shell.config.js]: Would it help to add this onto `header` as another type? Maybe:

```js
const header = {
  company: 'IBM',
  platform: '[Platform]',
  links: [
    {
      href: '/component/ui-shell--default',
      title: 'L1 link 1',
    },
    {
      href: '/component/ui-shell--default',
      title: 'L1 link 1',
    },
    {
      type: 'menu',
      links: [
        // ...
      ],
    }
  ]
};
```

Then the template could support either type, similar to what we'll need to do in the React component.

File: src/components/ui-shell/ui-shell.config.js [hunk 2]
```diff
@@ -35,6 +35,22 @@ const header = {
       title: 'Action 4',
     },
   ],
+  navlinks: [
```
Comment (joshblack) [inline src/components/ui-shell/ui-shell.config.js]: Could this be camel-case, e.g. `navLinks`?

**Review by joshblack — APPROVED** [review]

File: src/components/ui-shell/ui-shell.config.js [hunk 3]
```diff
@@ -135,6 +151,7 @@ module.exports = {
   },
   context: {
     header,
+    headernav,
```
Comment (asudoh) [inline src/components/ui-shell/ui-shell.config.js]: Cannot see `headernav` definition, which seems to have caused the Travis error:

```
[19:21:15] TypeError: Cannot read property 'forEach' of undefined
    at cache.get.then (/home/travis/build/IBM/carbon-components/tools/templates.js:100:21)
    at <anonymous>:null:null
```

The underlying `fractal.load()` fails to populate `componentSource` property in such case, and thus causes JavaScript error at https://github.com/IBM/carbon-components/blob/v9.22.0/tools/templates.js#L100.

**Comment by aledavila** [comment]
@asudoh good catch. Thanks try now

**Review by asudoh — COMMENTED** [review]
LGTM 👍 once our designer signs this off - Thanks @aledavila!

**Comment by designertyler** [comment]
The focus state allows the menus to overlap
screen shot 2018-09-27 at 11 12 26 am

**Review by designertyler — APPROVED** [review]

**Comment by chrisconnors-ibm** [comment]
does "remove focus on dropdown to prevent from staying open" affect keyboard navigation of menus

**Pull request merged**